### PR TITLE
Mainly Checkstyle violation fixes

### PR DESCRIPTION
**Pull request by artembilan**
- Upgrade to Checkstyle `7.1`
- Relax `RequireThis` rule a bit. Right now it does the effort only in case of overlapping. See https://github.com/checkstyle/checkstyle/issues/2362 for more info
- Enable some annotation rules and provide fixes for violations
- Enable `tabs indents` rule. This was the biggest fix in this PR
- Resolve `TODO` in the `MessagingMethodInvokerHelper` and fix tests to meet `IllegalStateException` now

### CODE REVIEW

**Comment by artembilan** [comment]
Well, looks like `?w=1` doesn't help so much with review :cry: 
Although it hides `69` files anyway :smile:

File: spring-integration-core/src/main/java/org/springframework/integration/support/utils/IntegrationUtils.java [hunk 1]
```diff
@@ -71,22 +71,21 @@ public static MessageBuilderFactory getMessageBuilderFactory(BeanFactory beanFac
...
 			catch (Exception e) {
 				if (logger.isDebugEnabled()) {
-					logger.debug("No MessageBuilderFactory with name '"
-								+ INTEGRATION_MESSAGE_BUILDER_FACTORY_BEAN_NAME
-								+ "' found: " + e.getMessage()
-								+ ", using default.");
+					logger.debug("No MessageBuilderFactory with name '" +
+							INTEGRATION_MESSAGE_BUILDER_FACTORY_BEAN_NAME +
+							"' found: " + e.getMessage() +
+							", using default.");
```
Comment (garyrussell) [inline spring-integration-core/src/main/java/org/springframework/integration/support/utils/IntegrationUtils.java]: What is this rule? My STS puts leading `+` when splitting strings.

Comment (artembilan) [inline spring-integration-core/src/main/java/org/springframework/integration/support/utils/IntegrationUtils.java]: https://github.com/spring-projects/spring-framework/wiki/Spring-Framework-Code-Style#line-wrapping:
>  Put the separator symbols at the end of the line rather on the next line 

But, eh, that was my own decision. 
I can come back to your preferences.
And we really can switch on http://checkstyle.sourceforge.net/config_whitespace.html#OperatorWrap checkstyle :smile:

**Comment by garyrussell** [comment]
Rebased, squashed, merged as 12fb590b95fc65e0aef7f75971cd510b9e9e000d

**Pull request closed without merging**